src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1508,14 +1509,18 @@
   returns a map associating alpha nodes with the facts they accept."
   [fact-type-fn ancestors-fn alpha-roots]
 
-  (let [ ;; If a customized fact-type-fn is provided,
+  (let [internal-fact? (if (compiling-cljs?)
+                         #(isa? (type %) ::eng/system-type)
+                         #(instance? ISystemFact %))
```

Comment:
You don't actually need this.  clara.rules.compiler/create-get-alphas-fn returns a (JVM) Clojure function object; this can't be called inside a JavaScript VM.  Clara's ClojureScript version uses another implementation of the get-alphas-fn at [1, 2].  The general principle underlying this is that ClojureScript code can use Clojure macros that return valid ClojureScript code since the ClojureScript compiler runs on the JVM.  In Clara's case, we have the defsession macro [3] that is referred to by ClojureScript code (in this case tests, but it would be done the same way in production code). [4]   However, this is only possible for raw ClojureScript code; other JVM objects can't be used.  This is why we have a significant amount of compilation code that is different between Clojure and ClojureScript.

The ClojureScript version of the get-alphas-fn doesn't have the wrapping of the get-fact-type-fn and ancestors-fn that it should; I logged https://github.com/cerner/clara-rules/issues/241 for that.

1. https://github.com/cerner/clara-rules/blob/0.13.0-RC4/src/main/clojure/clara/rules.cljc#L129
2. https://github.com/cerner/clara-rules/blob/0.13.0-RC4/src/main/clojure/clara/rules.cljc#L229
3. https://github.com/cerner/clara-rules/blob/0.13.0-RC4/src/main/clojure/clara/rules.cljc#L313
4. https://github.com/cerner/clara-rules/blob/0.13.0-RC4/src/test/common/clara/test_common.cljc#L8